crates/autopilot/src/infra/solvers/mod.rs

```diff
@@ -42,9 +42,31 @@ impl Driver {
         &self,
         request: &settle::Request,
         timeout: std::time::Duration,
-    ) -> Result<settle::Response> {
-        self.request_response("settle", request, Some(timeout))
+    ) -> Result<()> {
+        let url = util::join(&self.url, "settle");
+        tracing::trace!(
+            path=&url.path(),
+            body=%serde_json::to_string_pretty(request).unwrap(),
+            "solver request",
+        );
+
+        let response = self
+            .client
+            .post(url)
+            .json(request)
+            .timeout(timeout)
+            .send()
             .await
+            .context("send")?;
+        let status = response.status().as_u16();
+
+        tracing::trace!(%status, "solver response");
+
+        if status != 200 {
```

Comment:
Why not to use the custom types from `reqwest`?

```suggestion
        let status = response.status();
 
         tracing::trace!(%status, "solver response");
 
        if status != StatusCode::OK {
```